### PR TITLE
Setting element bugfix

### LINKED ISSUE #27875: System.Configuration.ConfigurationManager SettingElement Null Reference Exception
Related #27807 

While making the above tests, I discovered the default constructor for a SettingElement will throw a NRE if GetHashCode is called upon it.

Deemed as a bug, but what should it return? The same hashcode as a valid SettingElement? I don't have the background knowledge on what a 'proper' return value for this situation would be

### PR DESCRIPTION
**Pull request by Jlalond**
Fixes #27875

Checked to see if XmlNode.Value was null, and returning 0 if it was.

@danmosemsft Thoughts?

### CODE REVIEW

File: src/System.Configuration.ConfigurationManager/tests/System/Configuration/SettingElementTests.cs [hunk 1]
```diff
@@ -46,11 +46,11 @@ public void DefaultValueIsNull()
...
         [Fact]
-        public void DefaultSettingElementExceptionForGetHashCode()
+        public void DefaultConstructorEquality()
         {
-            var Element = new SettingElement();
-            Assert.Throws<NullReferenceException>(() => Element.GetHashCode());
-            //likely a bug
+            var ElementOne = new SettingElement();
+            var ElementTwo = new SettingElement();
+            Assert.True(ElementOne.Equals(ElementTwo));
```
Comment (danmosemsft) [inline src/System.Configuration.ConfigurationManager/tests/System/Configuration/SettingElementTests.cs]: This is a good test, but it's not testing your change. A test for your change would be `Assert.Equals(ElementOne.GetHashCode(), ElementTwo.GetHashCode());`

**Comment by danmosemsft** [comment]
As you see above this branch contains your earlier test additions, then you made another commit to make them vanish. At that point, one would typically use `git rebase -i`  to combine all the commits up to that point to eliminate any record of htem, since they are just mistakes.

As a rule we squash when we merge so they would not ultimately be in the product history, but the commit descriptions get retained so it's nice to clean up a bit.

Note once you've cleaned up your history locally, you'd need to do `git push -f` because you have rewritten history.

**Comment by Jlalond** [comment]
Yeah, I noticed afterwards I had made a mistake, thanks for taking the time to point it out and help me out!

**Comment by Jlalond** [comment]
Thanks for the tip on Rebase, I had never even thought about it, and because I've been a dev > 1 year, It's all still a crazy learning experience!

**Comment by danmosemsft** [comment]
Ah this test also runs against.NET Framework and of course it fails there as those bits don't have your fix. You will need to mark the failing test with an attribute to prevent it running against these bits. Search for SkipOnTargetFramework in other tests to see how this is done.

**Comment by Jlalond** [comment]
@danmosemsft Hey dan, added the attribute as requested

**Comment by danmosemsft** [comment]
Looks good thanks @Jlalond ! Going to do more?

**Comment by Jlalond** [comment]
@danmosemsft taking a slow pace, never worked (seen lol) a codebase this big, and I noticed the tests aren't friend assemblies to the tests, so just figuring it out as I go. Testing a code base I don't understand is bad practice imo, so researching and testing.

**Pull request merged**